src/BSON/Document.stub.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @generate-class-entries static
+ * @generate-function-entries static
+ */
+
+namespace MongoDB\BSON;
+
+final class Document implements \IteratorAggregate
+{
+    private function __construct() {}
+
+    final static public function fromBSONString(string $bson): Document {}
```

Comment:
Would `fromBSON` make more sense? We already have a string type for the parameter. If Document and ArrayList are going to be the only classes that represent BSON, I can't imagine there'd be any other `fromBSON*` method down the line.

Reply:
Good point, renamed.